lblod/reglement-publish-service
# Task model adjustments and cleanup

### Overview
This PR includes the following adjustments:
**Adjustments to `task` model**
- Usage of more generic `task` status URIs:
  * scheduled: 'http://redpencil.data.gift/id/concept/JobStatus/scheduled'
  * busy: 'http://redpencil.data.gift/id/concept/JobStatus/busy'
  * success: 'http://redpencil.data.gift/id/concept/JobStatus/success'
  * failed: 'http://redpencil.data.gift/id/concept/JobStatus/failed'
  * canceled: 'http://redpencil.data.gift/id/concept/JobStatus/canceled'
- Response of endpoints are now correctly structured following the `JSON:API` spec
- `template-publication` tasks are now linked to an `editor-document` instead of to the `document-container` => this ensures that these tasks are more uniquely identifiable.
**Clean-up of obsolete parts**
- Removal of obsolete `snippet-list-publication-tasks` endpoint + related functions

##### connected issues and PRs:
- https://github.com/lblod/app-reglementaire-bijlage/pull/88
- https://github.com/lblod/frontend-reglementaire-bijlage/pull/307

### Setup
Check-out https://github.com/lblod/app-reglementaire-bijlage/pull/88 for instructions

### How to test/reproduce
Check-out https://github.com/lblod/app-reglementaire-bijlage/pull/88 for instructions

### Challenges/uncertainties
- We should probably add an approach to check if there is already a task scheduled to publish a certain document.




### Checks PR readiness
